src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,200 @@
+complexity measure as developed in [^Rosso2013].
+Here, ``H_q``` can be the (q-order) Shannon-, Renyi or Tsallis
+entropy and ``Q_q`` based either on the Euclidean, Wooters, Kullback,
+q-Kullback, Jensen or q-Jensen distance as
+```math
+Q_q[P] = Q_q^0\\cdot D[P, P_e],
+```
+where ``D[P, P_e]`` is the distance between the obtained distribution ``P``
+and a uniform distribution with the same maximum number of bins, measured by
+the distance measure `dist`.
+
+## Usage
+
+The statistical complexity is exclusively used in combination with the related information measure
+(entropy).
+`complexity(c::StatisticalComplexity, x)` returns only the statistical complexity.
+The entropy can be accessed as a `Ref` value of the struct as
+```julia
+x = randn(100)
+c = StatisticalComplexity
+compl = complexity(c, x)
+entr = c.entr_val[]
+```
+To obtain both the entropy and the statistical complexity together as a `Tuple`, use the wrapper
+`entropy_complexity`(@ref).
+
+[^Rosso2007]: Rosso, O. A. et al. (2007). Distinguishing Noise from Chaos.
+    Physical Review Letters 99, no. 15: 154102. https://doi.org/10.1103/PhysRevLett.99.154102.
+[^Rosso2013]: Rosso, O. A. (2013) Generalized Statistical Complexity: A New Tool for Dynamical Systems.
+
+"""
+Base.@kwdef struct StatisticalComplexity{E, D, H} <: ComplexityEstimator
+    dist::D = JSDivergence()
+    est::E = SymbolicPermutation()
+    entr::H = Renyi()
+    entr_val::Base.RefValue{Float64} = Ref(0.0)
+end
+
+function complexity(c::StatisticalComplexity, x)
+    (; dist, est, entr) = c
+
+    p = probabilities(est, x)
+
+    H_q = entropy_normalized(entr, est, x)
+
+    L = total_outcomes(est, x)
+
+    # calculate distance between calculated distribution and uniform one
+    D_q = evaluate(dist, vec(p), fill(1.0/L, size(p)))
+
+    # generate distribution with just one filled bin
+    deterministic = zeros(size(p))
+    deterministic[1] = 1
+
+    D_max = evaluate(dist, deterministic, fill(1.0/L, size(p)))
+    C_q = D_q / D_max * H_q
+    c.entr_val[] = H_q
+
+    return C_q
+end
+
+"""
+    entropy_complexity(c::StatisticalComplexity, x)
+
+Calculate both the `StatisticalComplexity`(@ref) and corresponding entropy
+and return both as a `Tuple`.
+"""
+function entropy_complexity(c::StatisticalComplexity, x)
+    compl = complexity(c, x)
+   return (c.entr_val[], compl)
+end
+
+function complexity(c::StatisticalComplexity, p::Probabilities)
+    (; dist, est, entr) = c
+
+    L = total_outcomes(est)
+    norm = log(entr.base, L)
+    H_q = entropy(entr, p) / norm
+
+    # calculate distance between calculated distribution and uniform one
+    D_q = evaluate(dist, vec(p), fill(1.0/L, size(p)))
+
+    # generate distribution with just one filled bin
+    deterministic = zeros(size(p))
+    deterministic[1] = 1
+
+    D_max = evaluate(dist, deterministic, fill(1.0/L, size(p)))
+    C_q = D_q / D_max * H_q
+    c.entr_val[] = H_q
+
+    return C_q
+end
+
+function fill_probs_k!(p, prob_params, L, i, k)
```

Comment:
this function should be moved _after_ its called for the first time. Its name should also srtart with `_`.